apps/shell/test/allowlist.test.js

```diff
@@ -1,14 +1,66 @@
 /**
- * Allowlist helper function tests
+ * Helper functions tested
  */
 const helpers = require('../utils/helpers')
 
-/**
- * Constants used during test
- */
-const ALLOW_LIST = new Set(['owens.osc.edu', 'pitzer.osc.edu', '*.ten.osc.edu'])
+//generate_host_allowlist
+describe('Helper function generate_host_allowlsit', () => {
+  test('it should turn a colon-delimited string into a set', () => {
+    let allowlist = 'pitzer.osc.edu:owens.osc.edu:*.ten.osc.edu'
+    let host_allowlist = new Set(['pitzer.osc.edu', 'owens.osc.edu', '*.ten.osc.edu']);
+    expect(helpers.generate_host_allowlist(allowlist)).toMatchObject(host_allowlist);
+  })
+
+  test('it should return an empty set for an undefined allowlist', () => {
+    let allowlist;
+    let host_allowlist = new Set;
+    expect(helpers.generate_host_allowlist(allowlist)).toMatchObject(host_allowlist);
+  })
+})
+
+
+const owens = {host: 'owens.osc.edu', default: true};
+const pitzer = {host: 'pitzer.osc.edu', default: false};
+const ruby = {host: 'ruby.osc.edu', default: undefined}; // Testing case where default is NOT set
+//pitzer expansion has v2.metadata.hidden = true so it is not included
+const cluster_sshhosts = [owens, pitzer, ruby];
```

Comment:
These should not be defined here, rather in the tests themselves. Actually, even though many our tests are rife with the practice of setting up variables in a setup function or worse (see the atrocious rspec tests in ood_core), the best tests have no setup and are not DRY - rather, the assertion has everyting you need at a glance to understand what is happening.

For example:

```
expect(helpers.generate_cluster_sshhosts('./test/clusters.d')).toMatchObject([
  {host: 'owens.osc.edu', default: true},
  {host: 'pitzer.osc.edu', default: false},
  {host: 'pitzer.osc.edu', default: undefined}
]);
```

and

```
expect(helpers.generate_default_sshhost([{host: 'pitzer.osc.edu'}, {host: 'ruby.osc.edu'}])).toBe('pitzer.osc.edu')
```

it doesn't take much space to inline the expected values or actual values, and then those tests are far more readable. If they fail you know immediately why, without having to study the code.